jena-base/src/main/java/org/apache/jena/atlas/iterator/Iter.java

```diff
@@ -29,6 +29,14 @@
 import org.apache.jena.atlas.lib.Closeable ;
 import org.apache.jena.atlas.lib.Sink ;
 
+/**
+ * Iter provides utilities for working with {@link Iterator}s.
+ *
+ * Iter should never be used as a return type or parameter type in the public contract of a class. It is only to be used
+ * inside implementation code and is instantiated only to allow method-chaining as part of a calculation.
```

Comment:
IMO `Iter` is better than `ExtendedIterator` (more general, separated out) and should be used on general iterators. `ExtendedIterator` exists in jena-core and Graph API only and as a Triple iterator. 

As ExtendedIterator is exposed, it can't be changed. 

So if and where working on triple iterators out of Graph, use ExtendedIterator, else Iter.